lib/components/base/Promotion.vue

```diff
@@ -26,22 +26,22 @@
   </div>
 </template>
 
-<script setup>
+<script setup lang="ts">
 import { computed } from 'vue'
 import { BisectIcon } from '@'
 
-const props = defineProps({
-  external: {
-    type: Boolean,
-    default: true,
-  },
-  queryParam: {
-    type: String,
-    default: '',
-  },
-})
+const props = withDefaults(
+  defineProps<{
+    external: boolean
+    queryParam: string
+  }>(),
+  {
+    external: true,
+    queryParam: '',
+  }
+)
 
-const target = computed(() => (props.external ? '_blank' : '_self'))
+const target = computed<string>(() => (props.external ? '_blank' : '_self'))
```

Comment:
```suggestion
const target = computed(() => (props.external ? '_blank' : '_self'))
```

It should be able to automatically infer that this is a ~~`string`~~ `'_blank' | '_self'`.

Reply:
Done ^^.